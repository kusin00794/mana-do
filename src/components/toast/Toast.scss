@import '../../styles/_zIndex.scss';
@import '../../styles/variables.scss';

.toast {
  position: fixed;
  top: 30px;
  left: 50%;
  min-width: 250px;
  padding: 16px;
  border-radius: 2px;
  text-align: center;
  font-size: 16px;
  color: $white-color;
  z-index: $zIndexToast;
  visibility: hidden;
  transform: translate(-50%, 0);

  &.show {
    visibility: visible;
    animation: fadeIn 0.5s, fadeout 0.5s 2.5s;
  }

  &.danger {
    background-color: $danger-color;
  }

  &.warning {
    background-color: $warning-color;
  }

  &.info {
    background-color: $info-color;
  }

  &.success {
    background-color: $success-color;
  }
}

@-webkit-keyframes fadeIn {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 30px;
    opacity: 1;
  }
}

@keyframes fadeIn {
  from {
    top: 0;
    opacity: 0;
  }
  to {
    top: 30px;
    opacity: 1;
  }
}

@-webkit-keyframes fadeOut {
  from {
    bottom: 30px;
    opacity: 1;
  }
  to {
    bottom: 0;
    opacity: 0;
  }
}

@keyframes fadeOut {
  from {
    bottom: 30px;
    opacity: 1;
  }
  to {
    bottom: 0;
    opacity: 0;
  }
}
