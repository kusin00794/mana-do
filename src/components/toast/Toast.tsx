import React, { FC, useEffect, useState } from 'react';

import { IToast } from './Toast.d';
import './Toast.scss';

export const validToastType = {
  danger: 'danger',
  warning: 'warning',
  info: 'info',
  success: 'success',
};

const autoHideTime = 3000;

const Toast: FC<IToast> = ({ content, type, id }) => {
  const [isShow, setIsShow] = useState(false);

  useEffect(() => {
    if (id) {
      setIsShow(true);
      const interval = setInterval(() => {
        setIsShow(false);
      }, autoHideTime);
      return () => {
        clearInterval(interval);
      };
    }
  }, [id]);

  let toastType = '';
  switch (type) {
    case validToastType.danger:
      toastType = validToastType.danger;
      break;
    case validToastType.warning:
      toastType = validToastType.warning;
      break;
    case validToastType.info:
      toastType = validToastType.info;
      break;
    case validToastType.success:
      toastType = validToastType.success;
      break;
    default:
      toastType = validToastType.info;
      break;
  }

  return <div className={`toast${isShow ? ' show' : ''} ${toastType}`}>{content}</div>;
};

export default Toast;
