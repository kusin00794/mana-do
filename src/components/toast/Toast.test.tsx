import shortid from 'shortid';

import { testSnapshots } from '../../utils/test';
import Toast, { validToastType } from './Toast';

describe('<Toast />', () => {
  testSnapshots(Toast, [
    {
      props: { id: shortid(), content: validToastType.danger },
      description: 'default render',
    },
    {
      props: { id: shortid(), content: validToastType.danger, type: validToastType.danger },
      description: 'danger render',
    },
    {
      props: { id: shortid(), content: validToastType.info, type: validToastType.info },
      description: 'info render',
    },
    {
      props: { id: shortid(), content: validToastType.success, type: validToastType.success },
      description: 'success render',
    },
    {
      props: { id: shortid(), content: validToastType.warning, type: validToastType.warning },
      description: 'warning render',
    },
  ]);
});
