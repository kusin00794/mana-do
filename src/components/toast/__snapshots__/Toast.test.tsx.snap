// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> snapshots danger render 1`] = `
<div
  className="toast show danger"
>
  danger
</div>
`;

exports[`<Toast /> snapshots default render 1`] = `
<div
  className="toast info"
>
  danger
</div>
`;

exports[`<Toast /> snapshots info render 1`] = `
<div
  className="toast info"
>
  info
</div>
`;

exports[`<Toast /> snapshots success render 1`] = `
<div
  className="toast show success"
>
  success
</div>
`;

exports[`<Toast /> snapshots warning render 1`] = `
<div
  className="toast warning"
>
  warning
</div>
`;
